distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Request_Body.enso

```diff
@@ -1,35 +1,44 @@
-## The HTTP request body.
+import project.Any.Any
+import project.Data.Boolean.Boolean
+import project.Data.Map.Map
+import project.Data.Text.Encoding.Encoding
+import project.Data.Text.Text
+import project.System.File.File
+from project.Data.Boolean.Boolean import False, True
+
+## The HTTP POST request body.
 type Request_Body
 
-    ## Empty request body.
-    Empty
-
     ## Request body with text.
 
        Arguments:
        - text: The plain text in the request body.
-    Text text
+    Text (text:Text) (encoding:Encoding=Encoding.utf_8) (content_type:Text="text/plain")
 
-    ## Request body with JSON.
+    ## Request body with an object to be sent as JSON.
 
        Arguments:
-       - json: The JSON in the request body.
-    Json json
+       - x: The object to convert to JSON using `.to_json`.
+    Json (x:Any)
 
-    ## Request body with form data.
+    ## Request body with an object to be sent as a binary file.
 
        Arguments:
-       - form: The form data in the request body.
-    Form form
+       - file: The file to send.
+    Binary (file:File)
 
-    ## Request body with file data.
+    ## Request body with form data.
 
        Arguments:
-       - file: The file data in the request body.
-    File file
+       - form_data: the form fields (text or file) to be sent
+       - url_encoded: if true, use a URL-encoded form; otherwise, use a
+         multi-part encoding.
+    Form_Data (form_data:(Map Text (Text | File))) (url_encoded:Boolean=False)
 
-    ## Request body with binary.
+    ## Empty request body; used for GET
+    Empty
```

Comment:
Shouldn't we just use `Nothing` in that case?

Reply:
I would rather this be 100% an explicit choice, and not an accidental `Nothing`. That way, if you mean to pass something but accidentally produce Nothing, you'll know.

Reply:
Ok, sounds good indeed